dbcon/mysql/ha_mcs_impl.cpp

```diff
@@ -3086,6 +3086,10 @@ void ha_mcs_impl_start_bulk_insert(ha_rows rows, TABLE* table, bool is_cache_ins
             //@bug 6122 Check how many columns have not null constraint. columnn with not null constraint will not show up in header.
             unsigned int numberNotNull = 0;
 
+            // THere's a chance that a previously aborted query left garbage in ci->columnTypes
```

Comment:
Good catch. Unfortunately we can't just put a partial cleanup here. The clean up must happen inside ha_mcs::impl_external_lock(). If it is impossible we need to find out why it isn't fired up for aborted queries. This change calls for a separate issue IMHO.